test/bad-arguments.test.ts

```diff
@@ -0,0 +1,317 @@
+          });
+        });
+      });
+    });
+  });
+
+  PLUGINTYPE_DIRECTIVES.forEach((directive) => {
+    [directive, camelize(directive)].forEach((key) => {
+      describe(`${key } directive`, () => {
+        it('errors with an empty array', () => {
+          assertThrowsWithDirective(key, [], `${directive} must have at least one value. To block everything, set ${directive} to ["'none'"].`);
+        });
+
+        it('errors with a non-array', () => {
+          [
+            null,
+            undefined,
+            true,
+            {},
+            '',
+            'https://example.com',
+            function () {},
+          ].forEach((value) => {
+            assertThrowsWithDirective(key, value, `"${value}" is not a valid value for ${directive}. Use an array of strings.`);
+          });
+        });
+
+        it('errors when called with an array that contains non-strings', () => {
+          assertThrowsWithDirective(key, ['application/x-shockwave-flash', 420], '"420" is not a valid plugin type. Only non-empty strings are allowed.');
+        });
+
+        it('errors with unquoted values', () => {
+          assertThrowsWithDirective(key, ['none'], `"none" must be quoted in ${directive}. Change it to "'none'" in your source list. Force this by enabling loose mode.`);
+        });
+
+        it("errors when called with values that don't make sense, like 'self', 'unsafe-inline', and 'unsafe-eval'", () => {
+          assertThrowsWithDirective(key, ['self'], `"self" does not make sense in ${directive}. Remove it.`);
+          assertThrowsWithDirective(key, ['unsafe-inline'], `"unsafe-inline" does not make sense in ${directive}. Remove it.`);
+          assertThrowsWithDirective(key, ['unsafe-eval'], `"unsafe-eval" does not make sense in ${directive}. Remove it.`);
+          assertThrowsWithDirective(key, ["'self'"], `"'self'" does not make sense in ${directive}. Remove it.`);
+          assertThrowsWithDirective(key, ["'unsafe-inline'"], `"'unsafe-inline'" does not make sense in ${directive}. Remove it.`);
+          assertThrowsWithDirective(key, ["'unsafe-eval'"], `"'unsafe-eval'" does not make sense in ${directive}. Remove it.`);
+        });
+      });
+    });
+  });
+
+  URI_DIRECTIVES.forEach((directive) => {
+    [directive, camelize(directive)].forEach((key) => {
+      describe(`${key} directive, a URI`, () => {
+        it('errors when called with non-string values', () => {
+          [
+            null,
+            undefined,
+            {},
+            { length: 0 },
+            { length: 2 },
+            [],
+            ['example.com'],
+            123,
+            true,
+            '',
+          ].forEach((value) => {
+            assertThrowsWithDirective(key, value, `"${value}" is not a valid value for ${directive}. Use a non-empty string.`);
+          });
+        });
+      });
+    });
+  });
+
+  expect(REQUIRESRIFOR_DIRECTIVES).toStrictEqual(['require-sri-for']);
```

Comment:
Does this work outside of a Jest test? What happens if this fails?

Reply:
Have verified that this does work - if it fails, it appears as a failed test suite in the normal way (except isn't labelled with a custom test description). Could add a message if you think it's best?